src/platform/user/profile/vet360/actions/transactions.js

```diff
@@ -164,10 +165,22 @@ export function createTransaction(
         ? await apiRequest(route, options)
         : await localVet360.createTransaction();
 
-      recordEvent({
-        event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
-        'profile-section': analyticsSectionName,
-      });
+      // We want the validateAddreses method handling dataLayer events for saving / updating addresses.
+      if (vaProfileUseAddressValidation(state)) {
+        if (!fieldName.toLowerCase().includes('address')) {
+          recordEvent({
+            event:
+              method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
+            'profile-section': analyticsSectionName,
+          });
+        } else {
```

Comment:
this `else` should be attached to the outer `if (vaProfileUseAddressValidation(state))`